### DISCUSSION TITLE
wsl2 在使用 X11 时出错

### DISCUSSION THREAD

**Discussion by Light1110** [physics-data/faq#21] [Q&A] [answered]
出错信息如下：
已开启 VcXsrv，并配置环境变量 `DISPLAY=:0`
```
xeyes
Error: Can't open display: :0
```

**Comment by Light1110** [comment, marked as answer]
解决方案：
1. 如下配置环境变量 `DISPLAY`
```
export DISPLAY=$(awk '/nameserver / {print $2; exit}' /etc/resolv.conf 2>/dev/null):0
```
2. 配置 X11 server 使其允许远程接入
如：使用 VcXsrv 时，勾选 `Disable access control`   

参考：https://stackoverflow.com/questions/61110603/how-to-set-up-working-x11-forwarding-on-wsl2

**Comment by jiegec** [comment]
Thanks! The solution is merged to faq.
